crates/optimism/trie/src/prune/pruner.rs

```diff
@@ -72,20 +73,52 @@ where
         }
 
         // at this point `latest_block` is always greater than `min_block_interval`
-        let new_earliest_block = latest_block - self.min_block_interval;
+        let target_earliest_block = latest_block - self.min_block_interval;
 
         info!(
             target: "trie::pruner",
             from_block = earliest_block,
-            to_block = new_earliest_block,
+            to_block = target_earliest_block,
            "Starting pruning proof storage",
         );
 
-        let mut final_diff = BlockStateDiff::default();
-        // Fetch all diffs from (earliest_block + 1) to new_earliest_block (inclusive)
-        // initial proof data contains the state at `earliest_block`, so we start from
-        // earliest_block + 1
-        for i in (earliest_block + 1)..=new_earliest_block {
+        let mut current_earliest_block = earliest_block;
+        let mut prune_output = PrunerOutput {
+            start_block: earliest_block,
+            end_block: target_earliest_block,
+            ..Default::default()
+        };
+
+        // Prune in batches
+        while current_earliest_block < target_earliest_block {
+            // Calculate the end of this batch
+            let batch_end_block = std::cmp::min(
```

Comment:
```suggestion
            let batch_end_block = cmp::min(
```
pls import at least `cmp` module in the module prelude